generator/transit_generator.cpp

```diff
@@ -43,12 +45,17 @@ string GetFileName(string const & filePath)
 
 /// \brief Reads from |root| (json) and serializes an array to |serializer|.
 template <class Item>
-void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer)
+void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer,
```

Comment:
I'm not sure we really need to pass handlers to every instantiation. Perhaps it's better to create a separate function. Up to you.

Reply:
In case of separate method I'll have a significant amount of copypaste code:
```c++
template <class Item>
void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer)
{
  vector<Item> items;

  DeserializerFromJson deserializer(root.get());
  deserializer(items, key.c_str());

  serializer(items);
}

template <>
void SerializeObject<Gate>(my::Json const & root, string const & key, Serializer<FileWriter> & serializer)
{
  vector<Gate> items;

  DeserializerFromJson deserializer(root.get());
  deserializer(items, key.c_str());

  //Some more code to fill |m_pedestrianFeatureIds|

  serializer(items);
}
```

Reply:
Looks simpler to me since you don't need to keep track of the calls with non-trivial handler. Unless there are more opinions, up to you.